featherweb/__init__.py

```diff
@@ -35,56 +36,66 @@ def run(self, timeout=5, callback=None, **kwargs):
+                for fd, event in events:
+                    if event & select.POLLHUP or event & select.POLLERR:
+                        poller.unregister(self.m_Socket)
+                        raise Exception ("POLLHUP/POLLERR")
 
-                if id(fd) is not id(self.m_Socket) or not event & select.POLLIN:
-                    continue
+                    if id(fd) is not id(self.m_Socket) or not event & select.POLLIN:
+                        continue
+
+                    client, address = self.m_Socket.accept()
+
+                    try:
+                        f = client.makefile('rwb', 0)
 
-                client, address = self.m_Socket.accept()
+                        response = HTTPRequest(client, f.readline())
 
-                try:
-                    f = client.makefile('rwb', 0)
+                        # This may be dangerous for the ESP8266.  Request headers may be extensively large - a simple
+                        # Request with lots of HTTP headers could cause OOM crash. Request headers may be 8-16KB!
+                        while True:
+                            line = f.readline()
+                            if not line or line == b'\r\n':
+                                break
+                            k, v = line.split(b":", 1)
+                            response.headers[k] = v.strip()
 
-                    response = HTTPRequest(client, f.readline())
+                        found = False
+                        for e in self.m_Routes:
+                            pattern = e[0]
+                            handler = e[1]
 
-                    # This may be dangerous for the ESP8266.  Request headers may be extensively large - a simple
-                    # Request with lots of HTTP headers could cause OOM crash. Request headers may be 8-16KB!
-                    while True:
-                        line = f.readline()
-                        if not line or line == b'\r\n':
-                            break
-                        k, v = line.split(b":", 1)
-                        response.headers[k] = v.strip()
+                            if response.path.split('?', 1)[0] == pattern:
+                                found = True
+                                break
 
-                    found = False
-                    for e in self.m_Routes:
-                        pattern = e[0]
-                        handler = e[1]
+                        if not found:
+                            raise
 
-                        if response.path.split('?', 1)[0] == pattern:
-                            found = True
-                            break
+                        handler(response)
 
-                    if not found:
-                        raise
+                    except Exception as e:
+                        client.sendall('HTTP/1.0 404 NA\r\n\r\n')
 
-                    handler(response)
+                    except KeyboardInterrupt:
```

Comment:
Is KeyboardInterrupt unreachable due to general exception clause above?

Reply:
Oops! Fixed. :)